### PR TITLE
Add k3s upgrade script

### PR DESCRIPTION
**Pull request by zimme**
This adds a script to upgrade k3s separately from k3os.

/cc @bhale

### CODE REVIEW

**Comment by zimme** [comment]
There, should be working as expected just tried it on my dev board cluster and no problems with this latest version of the script

File: overlay/sbin/k3s-upgrade [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+# Take a parameter of the version number (i.e. v0.4.0) if it is given, otherwise use latest
+if [ -z $K3S_VERSION ]
+then
+    K3S_VERSION=$(curl -sL api.github.com/repos/rancher/k3s/releases/latest | jq .tag_name -r)
+fi
+
+if [ $PROC == "x86_64" ]
+then
+    ARCH="amd64"
```
Comment (bhale) [inline overlay/sbin/k3s-upgrade]: On AMD64, the k3s release binary is simply called 'k3s' with no arch suffix. In this case we should say something like.

# Special case for amd64 - k3s release has no arch suffix
ARCH=""

File: overlay/sbin/k3s-upgrade [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    ARCH="arm64"
+elif [ $PROC == "armv7l" ]
+then
+    ARCH="armhf"
+else
+    echo "Unsupported CPU architecture."
+    exit 1
+fi
+
+K3S_URL="https://github.com/rancher/k3s/releases/download/${K3S_VERSION}/k3s-${ARCH}"
```
Comment (bhale) [inline overlay/sbin/k3s-upgrade]: On amd64, the release binary is called 'k3s' with no arch suffix, so we need a special case to handle this.

`if [ $ARCH == "amd64" ]
then
     K3S_URL="https://github.com/rancher/k3s/releases/download/${K3S_VERSION}/k3s"
else
     K3S_URL="https://github.com/rancher/k3s/releases/download/${K3S_VERSION}/k3s-${ARCH}"
fi`

**Comment by zimme** [comment]
Will probably just include the - in the ARCH and set it to nothing for amd64 and remove the - from the hardcoded part of the url. That should not require any extra if statements

**Comment by dweomer** [comment]
Superseding functionality is being worked on at https://github.com/dweomer/system-upgrade-controller to be integrated into k3OS by https://github.com/dweomer/k3os/tree/system-upgrade-controller.

An upgrade Plan can be constructed to upgrade the entire system or one could write a Plan per component, in this case `k3s`. For manual upgrades, the `k3os upgrade` command will replace this script.

Thank you for the contribution @zimme and sorry for the lag on this one.

**Comment by dweomer** [comment]
See #348

**Pull request closed without merging**